### DISCUSSION TITLE
How to checkout a bare repository?

### DISCUSSION THREAD

**Discussion by Reverier-Xu** [GitoxideLabs/gitoxide#838] [Q&A] [answered]
Hi, thank you for your awesome work!

I want to checkout the head commit from the bare repository to some working folder, then I wrote the code referring to the method [here](https://github.com/Byron/gitoxide/blob/main/gix-worktree/tests/worktree/checkout.rs#L376):

```rust
    pub fn checkout_head(&self, dst_path: impl AsRef<Path>) -> anyhow::Result<()> {
        let dst_path = dst_path.as_ref();
        let git_path = self.path.clone();
        debug!("index path: {}", git_path.join("index").as_path().display());
        let mut index = gix_index::File::at(git_path.join("index"), Sha1, Default::default())?;
        let odb = gix::odb::at(git_path.join("objects"))?
            .into_inner()
            .into_arc()?;
        let _outcome = gix_worktree::checkout(
            &mut index,
            dst_path,
            move |oid, buf| odb.find_blob(oid, buf),
            &mut progress::Discard,
            &mut progress::Discard,
            &AtomicBool::default(),
            gix_worktree::checkout::Options {
                overwrite_existing: true,
                ..Default::default()
            },
        )?;
        Ok(())
    }
```

But it failed with message `An IO error occurred while opening the index`, it seems that bare repository does not have `index` file.

any other ways that could do this? thank you for you help!

**Comment by Reverier-Xu** [comment]
the method in git goes here:

```
git --work-tree=<working folder> --git-dir=<bare repository> checkout -f
```

**Comment by Byron** [comment, marked as answer]
Indices can be created on the fly by [using a tree object](https://docs.rs/gix-index/0.15.1/gix_index/struct.State.html#method.from_tree).

However, please note that at this time the checkout implementation is nothing more than experimental, a lot of the required implementation is missing.

Further, I wonder if `git worktree` would be suitable for your use-case. There is also `git archive` if detached worktrees are desired, even though there is no `gix` counterpart yet.

I hope that helps.
